src/util-device.c

```diff
@@ -60,14 +71,48 @@ int LiveGetOffload(void)
 /**
  *  \brief Add a pcap device for monitoring
  *
+ * To be used during option parsing. When a device has
+ * to be created during runmode init, use LiveRegisterDeviceFull()
+ *
  *  \param dev string with the device name
  *
  *  \retval 0 on success.
  *  \retval -1 on failure.
  */
 int LiveRegisterDevice(const char *dev)
 {
-    LiveDevice *pd = SCCalloc(1, sizeof(LiveDevice));
+    PreLiveDevice *pd = NULL;
+
+    pd = SCCalloc(1, sizeof(PreLiveDevice));
+    if (unlikely(pd == NULL)) {
+        return -1;
+    }
+
+    pd->dev = SCStrdup(dev);
+    if (unlikely(pd->dev == NULL)) {
+        SCFree(pd);
+        return -1;
+    }
+
+    TAILQ_INSERT_TAIL(&pre_live_devices, pd, next);
+
+    SCLogDebug("Device \"%s\" registered.", dev);
+    return 0;
+}
+
+/**
+ *  \brief Add a pcap device for monitoring and create structure
+ *
+ *  \param dev string with the device name
+ *
+ *  \retval 0 on success.
+ *  \retval -1 on failure.
+ */
+int LiveRegisterDeviceFull(const char *dev)
```

Comment:
I don't like this name. It's not self explaining and it's unclear why all callers need it.

Maybe it's better to have a special call for devices that need storage 'RegisterDeviceWithStorage' that are properly set up in Finalize

Reply:
Proposal to have the following changes:
- Regular LiveRegisterDevice become LiveRegisterDeviceName
- napatech and pfring use LiveRegisterDevice that do not change meaning

Reply:
This gives the following result: https://github.com/regit/suricata/commit/98b621e9424e6f8a336a042cddc48ebc17627de1

Reply:
I'm wondering if there is a need for 2 separate tracks here. Can't we just use the same logic for all users? So register only the name from the runmodes and then call the finalize to finish the setup? Right now it wouldn't be very clear to API consumers when to use one or the other.